plugins/systemadmin/systemadmin.plugin.zsh

```diff
@@ -140,12 +140,16 @@ d0() {
 
 # gather external ip address
 geteip() {
-    curl http://ifconfig.me
+    echo "$(curl -s -S http://ipecho.net/plain)"
 }
 
 # determine local IP address
 getip() {
-    ifconfig  | grep 'inet addr:'| grep -v '127.0.0.1' | cut -d: -f2 | awk '{ print $1}'
+    if [ $(hash | grep '^ip=') ]; then
```

Comment:
Ok, I understand you're using that to test whether the `ip` command is available. Instead, you can use the much shorter and preferred method: `if (( ${+commands[ip]} )); then ...`. See [an example](https://github.com/robbyrussell/oh-my-zsh/blob/1605d8264c6cfb7858bac96427a99f8bd9224a21/plugins/fbterm/fbterm.plugin.zsh#L3).

For one, `hash` on my machine doesn't output anything (you have to call `hash -v`). Secondly, you avoid a call to `grep` which is an external process and therefore requires forking a new process.